sim-lib/src/lib.rs

```diff
@@ -35,23 +35,78 @@ pub enum NodeId {
     Alias(String),
 }
 
+impl NodeId {
+    pub fn get_pk(&self) -> Result<&PublicKey, String> {
+        if let NodeId::PublicKey(pk) = self {
+            Ok(pk)
+        } else {
+            Err("NodeId is not a PublicKey".to_string())
+        }
+    }
+}
+
+impl std::fmt::Display for NodeId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                NodeId::PublicKey(pk) => pk.to_string(),
+                NodeId::Alias(a) => a.to_owned(),
+            }
+        )
+    }
+}
+
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct Config {
     pub nodes: Vec<NodeConnection>,
-    pub activity: Vec<ActivityDefinition>,
+    pub activity: Vec<ActivityParser>,
 }
 
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ActivityParser {
+    // The source of the payment.
+    #[serde(with = "serializers::serde_node_id")]
+    pub source: NodeId,
+    // The destination of the payment.
+    #[serde(with = "serializers::serde_node_id")]
+    pub destination: NodeId,
+    // The interval of the event, as in every how many seconds the payment is performed.
+    pub interval_secs: u16,
+    // The amount of m_sat to used in this payment.
+    pub amount_msat: u64,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ActivityDefinition {
     // The source of the payment.
     pub source: PublicKey,
     // The destination of the payment.
     pub destination: PublicKey,
     // The interval of the event, as in every how many seconds the payment is performed.
-    #[serde(alias = "interval_secs")]
     pub interval: u16,
     // The amount of m_sat to used in this payment.
-    pub amount_msat: u64,
+    pub amount: u64,
```

Comment:
Does it?

The only place we're constructing an `ActivityDefinition` instance is in main, and we need to parse it from `act` anyway:

```
activities.push(ActivityDefinition {
        source,
        destination,
        interval: act.interval_secs,
        amount: act.amount_msat,
});
```

Having said that, I don't really have a strong opinion for this, so I'm happy to change it if you do

Reply:
I think it makes sense to explicitly note when we're using msat/sat because both are represented with u64 and it's very easy to run into "off by 1000" errors in LN. 

>  Saves us a few line changes lower down as well.

This acts as a rename in `lib.rs`, so we have to change the logging in `produce_events` to just use `amount` ([around here](https://github.com/bitcoin-dev-project/sim-ln/pull/103/commits/47b3f5a2a62c0fe6cde0b4640eeee1de88a1bd42#diff-f891364aff98268ac6d94109bd61594c23af50e37e547fec9b0ea44a816459a0R566)).

Reply:
I also changed back `interval_secs` for consistency